src/ASM/FiniteElement.h

```diff
@@ -114,6 +114,8 @@ class MxFiniteElement : public FiniteElement
 protected:
   //! \brief Writes the finite element object to the given output stream.
   virtual std::ostream& write(std::ostream& os) const;
+  //! \brief Global Output stream operator.
+  friend std::ostream& operator<<(std::ostream& os, const MxFiniteElement& fe);
```

Comment:
Do you really need to overload this operator? Since it only invokes FiniteElement::write which is virtual, the correct behaviour will be achieved without it when invoked with a MxFiniteElement object as argument value.